diff -up libproxy-0.3.0/src/lib/url.c.nopassword-crash libproxy-0.3.0/src/lib/url.c
--- libproxy-0.3.0/src/lib/url.c.nopassword-crash	2014-06-17 10:24:02.260163932 -0400
+++ libproxy-0.3.0/src/lib/url.c	2014-06-17 10:28:54.074179750 -0400
@@ -290,6 +290,7 @@ px_url_new(const char *url)
 {
 	if (!url) return NULL;
 	const char *start = url;
+	const char *at, *colon, *slash;
 
 	/* Allocate pxURL */
 	pxURL *self  = px_malloc0(sizeof(pxURL));
@@ -300,12 +301,14 @@ px_url_new(const char *url)
 	start += strlen(self->scheme) + 3;
 
 	/* If we have a username and password */
-	if (strchr(start, '@') && (strchr(start, '/') > strchr(start, '@') || strchr(start, '/') == NULL))
+	at = strchr(start, '@');
+	colon = strchr(start, ':');
+	slash = strchr(start, '/');
+	if (at && colon && colon < at && (slash > at || slash == NULL))
 	{
-		if (!strchr(start, ':')) goto error; // Can't find user/pass delimiter
-		self->username = px_strndup(start, strchr(start, ':') - start);
+		self->username = px_strndup(start, colon - start);
 		start += strlen(self->username) + 1;
-		self->password = px_strndup(start, strchr(start, '@') - start);
+		self->password = px_strndup(start, at - start);
 		start += strlen(self->password) + 1;
 	}
 
