--- libproxy-0.3.0/src/lib/Makefile.am.sysconfdir	2009-08-14 15:00:32.000000000 -0400
+++ libproxy-0.3.0/src/lib/Makefile.am	2012-07-10 15:29:19.000000000 -0400
@@ -10,7 +10,7 @@
 						strdict.c strdict.h \
 						url.c url.h
 
-libproxy_la_CFLAGS  = -DMODULEDIR=\"@MODULEDIR@@DIR_SEPARATOR@\" -DSYSCONFDIR=\"@SYSCONFDIR@@DIR_SEPARATOR@\" -std=c99 -D_POSIX_C_SOURCE=200112L
+libproxy_la_CFLAGS  = -DMODULEDIR=\"@MODULEDIR@/\" -DSYSCONFDIR=\"@sysconfdir@/\" -std=c99 -D_POSIX_C_SOURCE=200112L
 libproxy_la_LDFLAGS = -lpthread -lm -no-undefined
 
 include_HEADERS     = proxy.h
--- libproxy-0.3.0/src/lib/Makefile.in.sysconfdir	2012-07-10 15:31:59.218464596 -0400
+++ libproxy-0.3.0/src/lib/Makefile.in	2012-07-10 15:32:08.443464508 -0400
@@ -277,7 +277,7 @@
 						strdict.c strdict.h \
 						url.c url.h
 
-libproxy_la_CFLAGS = -DMODULEDIR=\"@MODULEDIR@@DIR_SEPARATOR@\" -DSYSCONFDIR=\"@SYSCONFDIR@@DIR_SEPARATOR@\" -std=c99 -D_POSIX_C_SOURCE=200112L
+libproxy_la_CFLAGS = -DMODULEDIR=\"@MODULEDIR@/\" -DSYSCONFDIR=\"@sysconfdir@/\" -std=c99 -D_POSIX_C_SOURCE=200112L
 libproxy_la_LDFLAGS = -lpthread -lm -no-undefined
 include_HEADERS = proxy.h
 all: all-am
--- libproxy-0.3.0/src/modules/Makefile.am.sysconfdir	2009-08-14 15:06:30.000000000 -0400
+++ libproxy-0.3.0/src/modules/Makefile.am	2012-07-10 15:29:19.000000000 -0400
@@ -45,7 +45,7 @@
 
 # File Config Module
 config_file_la_SOURCES             = config_file.c
-config_file_la_CPPFLAGS            = -I$(top_srcdir)/src/lib -DSYSCONFDIR=\"@SYSCONFDIR@@DIR_SEPARATOR@\"
+config_file_la_CPPFLAGS            = -I$(top_srcdir)/src/lib -DSYSCONFDIR=\"@sysconfdir@/\"
 config_file_la_LIBADD              = ../lib/libproxy.la
 
 # GConf/GNOME Config Module
--- libproxy-0.3.0/src/modules/Makefile.in.sysconfdir	2012-07-10 15:31:32.502464856 -0400
+++ libproxy-0.3.0/src/modules/Makefile.in	2012-07-10 15:31:50.078464685 -0400
@@ -380,7 +380,7 @@
 
 # File Config Module
 config_file_la_SOURCES = config_file.c
-config_file_la_CPPFLAGS = -I$(top_srcdir)/src/lib -DSYSCONFDIR=\"@SYSCONFDIR@@DIR_SEPARATOR@\"
+config_file_la_CPPFLAGS = -I$(top_srcdir)/src/lib -DSYSCONFDIR=\"@sysconfdir@/\"
 config_file_la_LIBADD = ../lib/libproxy.la
 
 # GConf/GNOME Config Module
