diff -up libproxy-0.3.0/src/modules/pacrunner_webkit.c.webkit-crash libproxy-0.3.0/src/modules/pacrunner_webkit.c
--- libproxy-0.3.0/src/modules/pacrunner_webkit.c.webkit-crash	2009-08-14 15:00:33.000000000 -0400
+++ libproxy-0.3.0/src/modules/pacrunner_webkit.c	2014-06-23 16:32:01.237368208 -0400
@@ -197,6 +197,7 @@ _run(pxPACRunnerModule *self, pxPAC *pac
 	if (ctxs && strcmp(ctxs->pac, px_pac_to_string(pac)))
 	{
 		ctxs_free(ctxs);
+		((pxWebKitPACRunnerModule *) self)->ctxs = NULL;
 		ctxs = NULL;
 	}
 
@@ -224,6 +225,7 @@ _run(pxPACRunnerModule *self, pxPAC *pac
 error:
 	if (str) JSStringRelease(str);
 	ctxs_free(ctxs);
+	((pxWebKitPACRunnerModule *) self)->ctxs = NULL;
 	return tmp;
 }
 
